pages/index.tsx

```diff
@@ -10,32 +11,50 @@ import { AUTH_COOKIE_NAME } from 'shared/constants';
 
 interface Props {
   isAuthed: boolean;
+  seenOnBoarding: boolean;
 }
 
 const Index: NextPage<Props> = props => {
   const [isAuthed, setIsAuthed] = React.useState(props.isAuthed);
+  const [seenOnBoarding, setSeenOnBoarding] = React.useState(props.seenOnBoarding);
+  const router = useRouter();
+
+  // to avoid this error:
+  // During Pre-rendering (SSR or SSG) you tried to access a router method push, replace, back, which is not supported.
+  React.useEffect(() => {
```

Comment:
What's being done to avoid the error?  `React.useEffect` only runs clientside so this should behave as expected?  What was being attempted before?

Reply:
I tried `router.push('/onBoarding');` without `useEffect` hook first and got the error. So I added the comment about the error because i thought it was a workaround but if its normal ill remove it

Reply:
no longer needed, removed

Reply:
To clarify, the error is expected in the sense that routing logic shouldn't be happening at the top level of the component. The main body of a functional component is evaluated on every render.